src/controllers/user.js

```diff
@@ -176,6 +209,67 @@ const User = {
 
     return res.status(200).send({ imgURL });
   },
+
+  verify: async (req, res, next) => {
+    const { id } = req.params;
+    const { verificationCode } = req.body;
+    let result;
+
+    try {
+      result = await UserModel.findOneAndUpdate({ _id: id, verificationCode },
+        { verified: true }).exec();
+    } catch (err) {
+      return next(new APIError());
+    }
+
+    if (!result) {
+      return next(new APIError(
+        'User could not be found',
+        `User with id ${id} was not found`,
+        404,
+        `users/${id}/verify`,
+      ));
+    }
+
+    return res.status(200).send(result.toJSON());
+  },
+
+  emailPasswordRecovery: async (req, res, next) => {
+    const { id } = req.params;
+    let result;
+    const verificationCode = uuidv4();
+    try {
+      result = await UserModel.findOneAndUpdate({ _id: id }, verificationCode);
```

Comment:
Reusing verificationCode will wipe out a previous verificationCode that may exist. In the scenario where a non-verified user tries to reset their password, the link in their verification email will no longer work since the code will be different.

Reply:
Just leaving this as a comment bc its a super niche thing for our project